src/components/recordedit/recordedit.tsx

```diff
@@ -350,56 +447,258 @@ const RecordeditInner = ({
-      // but we cannot go basd on visible columns since some of these data might be for invisible fks.
-      reference.activeList.allOutBounds.forEach((col: any) => {
-        // copy the foreignKeyData (used for domain-filter support in foreignkey-field.tsx)
-        foreignKeyData.current[`c_${formValue}-${col.RID}`] = simpleDeepCopy(foreignKeyData.current[`c_${lastFormValue}-${col.RID}`]);
-
-        // copy the raw data (submitted to ermrestjs)
-        col.foreignKey.colset.columns.forEach((col: any) => {
-          const val = tempFormValues[`c_${lastFormValue}-${col.RID}`];
-          if (val === null || val === undefined) return;
-          tempFormValues[`c_${formValue}-${col.RID}`] = val;
+      // but we cannot go based on visible columns since some of this data might be for invisible fks.
+      tempFormValues = setOutboundForeignKeyValues(tempFormValues, formValue, lastFormValue);
+    }
+
+    return { tempFormValues, lastFormValue };
+  }
+
+  /**
+   * set values in foreignkey data and formValues for all out foreign key columns
+   *
+   * @param formValues the existing values in the form
+   * @param formNumber the form number we are setting values for
+   * @param lastFormValue the last form number that values are copied from
+   * @param checkPrefill if prefill should be checked for copying
+   * @returns updated form values to set in react hook form
+   */
+  const setOutboundForeignKeyValues = (formValues: any, formNumber: number, lastFormValue: number, checkPrefill?: boolean) => {
+    const tempFormValues = { ...formValues };
+    reference.activeList.allOutBounds.forEach((col: any) => {
+      const bulkFKObject = reference.bulkCreateForeignKeyObject;
+        // don't copy the value for the leaf column for an assoication that is unique
+      if (bulkFKObject?.isUnique && col.name === bulkFKObject.leafColumn.name) return;
+
+      // copy the foreignKeyData (used for domain-filter support in foreignkey-field.tsx)
+      foreignKeyData.current[`c_${formNumber}-${col.RID}`] = simpleDeepCopy(foreignKeyData.current[`c_${lastFormValue}-${col.RID}`]);
+
+      if (checkPrefill) {
+        // check prefill object for the columns that are being prefilled to update the new forms since we aren't calling getPrefilledDefaultForeignKeyData()
+        if (prefillObject?.fkColumnNames.indexOf(col.name) !== -1) {
+          tempFormValues[`c_${formNumber}-${col.RID}`] = formValues[`c_${lastFormValue}-${col.RID}`];
+        }
+      }
+
+      // copy the raw data (submitted to ermrestjs)
+      col.foreignKey.colset.columns.forEach((col: any) => {
+        const val = formValues[`c_${lastFormValue}-${col.RID}`];
+        if (val === null || val === undefined) return;
+
+        tempFormValues[`c_${formNumber}-${col.RID}`] = val;
+      });
+    });
+
+    return tempFormValues;
+  }
+
+  // show the prefill bulk foreign key modal if we have a prefill object and bulk foreign key recordset props
+  const openBulkForeignKeyModal = () => {
```

Comment:
I think you should modify this function to do everything that you're doing in the initial useEffect. And then just call that as well. You're doing the exact same thing as that useEffect here but some stuff are missing (which I don't think is intentional. For example the `domainRef` change that you've done above is missing from here).